src/backend/parser/analyze.c

```diff
@@ -1440,7 +1440,7 @@ transformSelectStmt(ParseState *pstate, SelectStmt *stmt)
 	qry->commandType = CMD_SELECT;
 
 	/* process the WITH clause independently of all else */
-	if (stmt->withClause)
+	if (stmt->withClause && !stmt->isPivot)
```

Comment:
should we first check for hook here ? we can probably reuse transform_pivot_clause_hook

Reply:
If in that case, then we will skip the intoClause check.

Reply:
I didn't mean actually calling the hook here. Just a safety check. I was suggesting something like
```if (stmt->withClause && (transform_pivot_clause_hook == NULL || (dialect == tsql && !stmt->isPivot))```
This just provides surety that this change does not come into picture if babelfish isn't installed.

Reply:
Thanks, will update here